### ISSUE TITLE
Have GitLab Omnibus set default email settings for Mattermost install

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#7520]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/PLT-7715)

This is a code change in GitLab Omnibus: https://gitlab.com/gitlab-org/omnibus-gitlab

With GitLab install, email works out of the box using "sendmail". However, in Mattermost email won't automatically work and requires the admin to configure it specifically for Mattermost: https://gitlab.com/gitlab-org/omnibus-gitlab/issues/2779

Proposal is to have GitLab Omnibus set default email settings for Mattermost install

This should be possible by adding some handling in the Mattermost recipe (https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/files/gitlab-cookbooks/gitlab/recipes/mattermost.rb) to take GitLab's SMTP settings (see https://docs.gitlab.com/omnibus/settings/smtp.html) if the default values are just "nil".

**Comment by bobashforth** [comment]
Jason, I'd like to take this one... let me know if/when it's approved.

**Comment by bobashforth** [comment]
@esethna, it looks like this path is obsolete:

https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/files/gitlab-cookbooks/gitlab/recipes/mattermost.rb.

The closest I can come to the referenced file is this, I think:

https://gitlab.com/bobashforth/omnibus-gitlab/blob/master/files/gitlab-cookbooks/mattermost/recipes/enable.rb

Can you confirm/correct as appropriate?

Thanks-

     Bob

**Comment by esethna** [comment]
Hi @bobashforth, sorry for the delay in answering your question. I'm adding @hmhealey to help clarify.

**Comment by hmhealey** [comment]
Hi @bobashforth,

The correct file is now https://gitlab.com/bobashforth/omnibus-gitlab/blob/master/files/gitlab-cookbooks/mattermost/attributes/default.rb#L110. The values set there will be written into the config.json.

**Comment by hmhealey** [comment]
@jasonblais Once this is done, we should reach out to GitLab to add those settings to [this list](https://docs.gitlab.com/omnibus/gitlab-mattermost/#upgrading-gitlab-mattermost-from-versions-prior-to-11-0) so that they remain supported in GitLab 11. I don't know if that involves just changing that in the documentation or if there's somewhere in code that we'd need to change it as well

**Comment by jasonblais** [comment]
@hmhealey Noted

**Comment by KuSh** [comment]
Seems a little bit more difficult than "Easy" ^^

Since the creation of the ticket gitlab omnibus switched to using environment variables to autoconfigure services and unfortunately in mattermost ["environment variables override settings in config.json"](https://docs.mattermost.com/administration/config-settings.html#configuration-settings) so autoconfiguring mattermost means disabling configuration through config.json / System Console

See https://gitlab.com/gitlab-org/omnibus-gitlab/-/issues/2779 for reference

**Comment by esethna** [comment]
@hmhealey is this ticket still valid?

**Comment by hmhealey** [comment]
I think it's still valid, but I agree that it's not easy. It's also a bit different than the rest of these since it requires changes to GitLab Omnibus and nothing to do with our code base.

Having them disabled in the System Console is fine since that's how the other settings configured by GitLab act. The only potential problem I could see is that users might have their email settings configured differently currently, so this change would potentially overwrite those. I don't think there's anything in Omnibus's configuration code that would let us not overwrite users who have manually configured email at this point like how we support different defaults for new installs vs upgrades.

**Comment by jasonblais** [comment]
@hmhealey to confirm, closing in favor of https://gitlab.com/gitlab-org/omnibus-gitlab/-/issues/2779 ?

**Comment by hmhealey** [comment]
Yeah, I think so. This is something we'd like, but it's not really ours to dictate.
